### PR TITLE
Utils: New - Extract WP export functionality to class. ED-2059

### PR DESCRIPTION
**Pull request by iNewLegend**

### CODE REVIEW

File: core/utils/wp-exporter.php [hunk 1]
```diff
@@ -0,0 +1,582 @@
...
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/*
+ * Originally made by WordPress.
+ *
+ * What changed:
+ *  Remove echos.
+ *  Remove filters.
```
Comment (matipojo) [inline core/utils/wp-exporter.php:13]: We need the filters back.

File: core/utils/wp-exporter.php [hunk 2]
```diff
@@ -0,0 +1,582 @@
...
+class WP_Exporter {
+	const WXR_VERSION = '1.2';
+
+	private static $default_args = [
+		'content'    => 'all',
+		'author'     => false,
+		'category'   => false,
+		'start_date' => false,
+		'end_date'   => false,
+		'status'     => false,
```
Comment (matipojo) [inline core/utils/wp-exporter.php:31]: PHPCS

Comment (matipojo) [inline core/utils/wp-exporter.php:31]: Where is the limit & offset?

File: tests/phpunit/elementor/core/utils/test-wp-exporter.php [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+		// Arrange.
+		require ABSPATH . 'wp-admin/includes/export.php';
+
+		$this->expected_errors( 'Cannot modify header information' );
+
+		ob_start();
+		export_wp();
+		$original_wp_export_output = ob_get_clean();
+
+		$clean_wp_export_output = $this->remove_non_require_content( $original_wp_export_output );
```
Comment (matipojo) [inline tests/phpunit/elementor/core/utils/test-wp-exporter.php:63]: Why not compare with all content?

**Review by matipojo — CHANGES_REQUESTED** [review]
Nice work!

File: core/utils/wp-exporter.php (hunk 1 above)
Comment (iNewLegend) [reply core/utils/wp-exporter.php]: Handled.

File: core/utils/wp-exporter.php (hunk 2 above)
Comment (iNewLegend) [reply core/utils/wp-exporter.php]: Handled.

File: tests/phpunit/elementor/core/utils/test-wp-exporter.php (hunk 3 above)
Comment (iNewLegend) [reply tests/phpunit/elementor/core/utils/test-wp-exporter.php]: Handled.

**Review by matipojo — CHANGES_REQUESTED** [review]
Fix PHPUnit failure.

File: core/utils/wp-exporter.php [hunk 4]
```diff
@@ -0,0 +1,680 @@
...
+		<wp:base_site_url>$wxr_site_url</wp:base_site_url>
+		<wp:base_blog_url>$rss_info_url</wp:base_blog_url>
+$dynamic
+	</channel>
+</rss>
+EOT;
+		return $result;
+	}
+
+	public function run() {
```
Comment (matipojo) [inline core/utils/wp-exporter.php:584]: Move to before private functions.

Comment (matipojo) [inline core/utils/wp-exporter.php:584]: Add PHPDoc for all methods.

**Review by matipojo — CHANGES_REQUESTED** [review]

File: core/utils/wp-exporter.php [hunk 5]
```diff
@@ -0,0 +1,680 @@
...
+
+		return $result;
+	}
+
+	/**
+	 * @param $post_ids
+	 *
+	 * @return string
+	 */
+	public function wxr_posts_list( $post_ids ) {
```
[the remark this answers is not in the document]
Comment (iNewLegend) [reply core/utils/wp-exporter.php]: Handled.

File: core/utils/wp-exporter.php (hunk 4 above)
Comment (iNewLegend) [reply core/utils/wp-exporter.php]: Handled.

Comment (iNewLegend) [reply core/utils/wp-exporter.php]: Handled.

**Review by matipojo — DISMISSED** [review]

**Pull request closed without merging**